### DISCUSSION TITLE
How can we use AddressSanitizer with test files?

### DISCUSSION THREAD

**Discussion by ytrewq271828** [ANLAB-KAIST/KENSv3#206] [Project Q&A 2. Transport Layer (TCP)]
Hello. I want to ask how can we use AddressSanitizer with KENS.

I tried to compile with gcc using -fsanitize=address option.
But the compilation has failed, maybe due to not specifying other linked files.

The material on using AddressSanitizer in the github wiki page didn't work for me.
Any guidance would be grateful.

**Comment by Pusnow** [comment]
Would you let me know which environment (OS, distro, kernel, compiler, compiler version, etc.) you are using?

**Reply by ytrewq271828** [reply]
I'm using WSL with Ubuntu 20.04.2 and gcc 9.4.0

**Reply by Pusnow** [reply]
Then

```
$ mkdir build && cd build
$ cmake -DSANITIZER=address ..
$ make -j
```

would work for you.

What compilation error messages do you get?

**Reply by ytrewq271828** [reply]
Yesterday I got some compilation error messages like the following.

c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.

Today the command "make -j" builds but it seems that there are no changes or creations of files.

**Reply by Pusnow** [reply]
> c++: fatal error: Killed signal terminated program cc1plus

It is probably due to small available memory (RAM).
Try recompilation with `make -j1` (it will take longer then `-j`).

**Reply by ytrewq271828** [reply]
Is it normal to finish build operation without creating files or changing the existing ones?

**Reply by Pusnow** [reply]
> Is it normal to finish build operation without creating files or changing the existing ones?

If there are precompiled files and source code is not modified.
Or some build script errors.

Try to rebuild from scratch by removing `build` and run `cmake` command again.
